### ISSUE TITLE
Obsidian - The RedstoneCoreOracle has a constant stale price threshold, this is dangerous to use with tokens that have a smaller threshold as the oracle will report stale prices as valid

### ISSUE THREAD

**Issue by sherlock-admin3** [sherlock-audit/2024-08-sentiment-v2-judging#126]
Obsidian

Medium

# The RedstoneCoreOracle has a constant stale price threshold, this is dangerous to use with tokens that have a smaller threshold as the oracle will report stale prices as valid

### Summary

Different tokens have different `STALE_PRICE_THRESHOLD`. The protocol uses a constant `STALE_PRICE_THRESHOLD = 3600` for all tokens in the RedstoneCoreOracle. 

The issue arises when the token actually has a STALE_PRICE_THRESHOLD < 3600, then the oracle will report the stale price as valid. 

Here are some tokens whose redstone priceFeed has a STALE_PRICE_THRESHOLD < 3600 (1 hour)

1. TRX/USD 10 minutes
2. BNB/USD 1 minute

### Root Cause

using a [constant](https://github.com/sherlock-audit/2024-08-sentiment-v2/blob/25a0c8aeaddec273c5318540059165696591ecfb/protocol-v2/src/oracle/RedstoneOracle.sol#L19) `STALE_PRICE_THRESHOLD = 3600`, rather than setting one for each token

### External pre-conditions

Token has a threshold < 3600

### Impact

The protocol will report stale prices as valid, this results in collateral being valued using stale prices.

It will lead to unfair liqudiations due to stale price valuation of collateral AND/OR a position not being liquidated due to stale price valuation of collateral

It will also lead to borrowing a wrong amount due to stale price valuation of collateral

### Mitigation

Set a unique `STALE_PRICE_THRESHOLD` for each token, similar to the chainlink oracle

**Comment by 0xSpearmint1** [comment]
escalate

The lead judge states that since each asset has it's own instance of the oracle the param can be changed

1. Firstly, the STALE_PRICE_THRESHOLD is a constant variable that is already set, there is no evidence that the team intended to change the currently set constant variable.

2. Secondly, since each oracles instance uses 2 price feeds to determine the USD price of the asset (Asset/ETH and ETH/USD), as long as the asset has a different threshold to ETH the described issue in the report will occur.

**Comment by sherlock-admin3** [comment]
> escalate
> 
> The lead judge states that since each asset has it's own instance of the oracle the param can be changed
> 
> 1. Firstly, the STALE_PRICE_THRESHOLD is a constant variable that is already set, there is no evidence that the team intended to change the currently set constant variable.
> 
> 2. Secondly, since each oracles instance uses 2 price feeds to determine the USD price of the asset (Asset/ETH and ETH/USD), as long as the asset has a different threshold to ETH the described issue in the report will occur.

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by ruvaag** [comment]
I think this should be a low because the intended behavior in the described case would be to use the worst stale price threshold which should mitigate this

**Comment by 0xSpearmint1** [comment]
What do you mean by the worst stale price? @ruvaag 

If you mean the smaller one, this will cause a serious DOS issue for the token with a larger threshold (liquidations will revert).

If you mean the larger one, this will allow borrowing the other token at a stale price.

The only mitigation is to have a seperate threshold for each token.

**Comment by cvetanovv** [comment]
I agree that the constant `STALE_PRICE_THRESHOLD` is not good to be hardcoded to 1 hour because each token pair has a different stale period when it needs to be updated. 

Because of this, I agree that this issue is more of a Medium because the price may be outdated.

I plan to accept the escalation and make this issue a Medium severity.

**Comment by 0xSpearmint1** [comment]
Hi @cvetanovv [#346](https://github.com/sherlock-audit/2024-08-sentiment-v2-judging/issues/346) is **not** a duplicate of this issue, it is actually invalid.

This issue is about using a constant `STALE_PRICE_THRESHOLD`

[#346](https://github.com/sherlock-audit/2024-08-sentiment-v2-judging/issues/346) describes a totally different attack vector which claims that the threshold is too short (threshold is set by the admin).

**Comment by HHK-ETH** [comment]
Agree, [346](https://github.com/sherlock-audit/2024-08-sentiment-v2-judging/issues/346) is similar but incomplete. It only talks about the max duration threshold and not the constant itself. As it was correctly pointed out it could also be an issue to use a duration too small.

It should be removed from duplicates 👍

**Comment by cvetanovv** [comment]
@0xspearmint1 @HHK-ETH Thanks for noting that #346 is not a duplicate of this issue. And it indeed uses a different attack vector.

My decision is to accept the escalation and make this issue and its duplicates Medium severity without #346.

**Comment by WangSecurity** [comment]
Result:
Medium
Has duplicates

**Comment by sherlock-admin4** [comment]
Escalations have been resolved successfully!

Escalation status:
- [0xspearmint1](https://github.com/sherlock-audit/2024-08-sentiment-v2-judging/issues/126/#issuecomment-2355952938): accepted
